docs/BackingServiceBestPractices.md

```diff
@@ -1,169 +1,286 @@
-    displayName: DB IP address
-    path: dbConnectionIP
-    x-descriptors:
-      - binding:env:attribute
-```
 
-### Operator Not Providing Metadata
 
-This feature enables operators that manage backing services but which don't
-have any metadata in their CSV to use the Service Binding Operator to bind
-together the service and applications. The Service Binding Operator binds all
-sub-resources defined in the backing service CR by populating the binding
-secret with information from Routes, Services, ConfigMaps, and Secrets owned
-by the backing service CR.
+### Scenarios
 
-[This is how resource and sub-resource relationships are set in
-Kubernetes.](https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/#owners-and-dependents)
 
-The binding is initiated by the introduction of this API option in the backing service CR:
-``` yaml
-detectBindingResources : true
-```
-When this API option is set to true, the Service Binding Operator
-automatically detects Routes, Services, ConfigMaps, and Secrets owned by
-the backing service CR.
+1. #### Use everything from the Secret  “status.data.dbCredentials”
+
+    Requirement : *Extract an entire configmap/Secret from the Kubernetes resource*
+
+
+    Annotation:
+
+    ```
+    “servicebinding.dev/dbcredentials”:”path={.status.data.dbcredentials},objectType=Secret”
+    ```
+
+
+    Descriptor:
+
+    ```
+    - path: data.dbcredentials
+      x-descriptors:
+        - urn:alm:descriptor:io.kubernetes:Secret 
+        - servicebinding
+    ```
+
+
+2. #### Use everything from the ConfigMap “status.data.dbConfiguration”
+
+
+    Requirement : *Extract an entire configmap/Secret from the Kubernetes resource*
+
+    Annotation
+
+    ```
+    “servicebinding.dev/dbConfiguration”: "path={.status.data.dbConfiguration},objectType=ConfigMap”
+    ```
+
+
+    Descriptor
+
+    ```
+    - path: data.dbConfiguration
+      x-descriptors:
+        - urn:alm:descriptor:io.kubernetes:ConfigMap 
+        - servicebinding
+    ```
+
+3. #### Use “certificate” from the ConfigMap “status.data.dbConfiguration” as an environment variable
+
+    Requirement : *Extract a specific field from the configmap/Secret from the Kubernetes resource and use it as an environment variable.*
+
+
+    Annotation
+
+    ```
+    “servicebinding.dev/certificate”:
+    "path={.status.data.dbConfiguration},objectType=ConfigMap"
+    ```
+
+
+    Descriptor
+
+
+    ```
+    - path: data.dbConfiguration
+      x-descriptors:
+        - urn:alm:descriptor:io.kubernetes:ConfigMap
+        - servicebinding:certificate:bindAs=envVar
+    ```
+
+
+4. #### Use “certificate” from the ConfigMap “status.data.dbConfiguration” as a volume mount
+
+    Requirement : *Extract a specific field from the configmap/Secret from the Kubernetes resource and use it as a volume mount.*
+
+
+    Annotation
+
+    ```
+    “servicebinding.dev/certificate”:
+    "path={.status.data.dbConfiguration},bindAs=volume,objectType=ConfigMap"
```

Comment:
Is `bindAs=volume` being supported in this PR?

Reply:
no.